woocommerce/woocommerce-ios
# Store creation M2 - domain selection: basic UI integration

Part of #8045 


### Description


Domain selection is the first step that I'm implementing for store creation, because profiler questions might be skipped due to API availability (we'll likely have them ready by the end of November and we're okay with skipping profiler questions in M2). This PR includes barebone UI using the existing components like `SearchHeader` and `List`, with many design improvements needed as a later subtask to match the latest design since the PR is already big. The empty and loading state design is still TBD and will be implemented in a future subtask.

Major changes include:
- Added a feature flag `storeCreationM2` and show the domain selector as the first step of store creation when the feature flag is enabled in `StoreCreationCoordinator`
- Created two SwiftUI views: `DomainSelectorView` and `DomainRowView` with their own view model
  - `DomainSelectorHostingController` was added as the UIKit wrapper for `DomainSelectorView`
  - In `DomainSelectorViewModel`, it loads domain suggestions from the API and updates the observable domains

### Testing instructions


- Log in to the app if needed
- Go to the Menu tab
- Tap `Switch store`
- Tap `+ Add a store`
- Tap `Create a new store` in the action sheet --> the store picker should be dismissed, then the domain selector should be shown with a very basic UI
- Enter some text to the field --> some domain suggestions should be loaded, and the search query should be highlighted in the domain attributed string
- Select a domain --> the substring that doesn't match the search query should be bold

---
- [ ] @jaclync tests the store creation flow when the feature is disabled

### Screenshots


one-word query | two-word query | two-word query - light mode
-- | -- | --
![Simulator Screen Shot - iPhone 14 - 2022-11-09 at 16 59 10](https://user-images.githubusercontent.com/1945542/200785973-59aff126-6cc6-4743-93e0-ae1dd4c5560f.png) | ![Simulator Screen Shot - iPhone 14 - 2022-11-09 at 16 59 32](https://user-images.githubusercontent.com/1945542/200785992-1566d796-10d6-4baa-a686-171f068bbbeb.png) | ![Simulator Screen Shot - iPhone 14 - 2022-11-09 at 16 59 41](https://user-images.githubusercontent.com/1945542/200786001-0ea9447c-d91a-4c98-bb82-cfdef88adfa0.png)

---
